xarray/core/iterators.py

```diff
@@ -11,9 +14,9 @@ class AbstractIter(abc.Iterator):
     def __init__(
         self,
         node: Tree,
-        filter_: Optional[Callable] = None,
-        stop: Optional[Callable] = None,
-        maxlevel: Optional[int] = None,
+        filter_: Callable | None = None,
+        stop: Callable | None = None,
+        maxlevel: int | None = None,
     ):
         """
         Iterate over tree starting at `node`.
```

Comment:
Odd format of this docstring, xarray usually follows the numpy docstring format.

Adding some examples how it's used would be nice as well. Then you would also get some doctests for free.

Reply:
I've migrated this documentation string up to the class level, which seems in keeping with the rest of the repository. I've then added examples for the subclasses of this abstract base class to those subclasses themselves.